syntax = "proto3";

package ai_processor;

service AiService
{
    rpc AutoGeneratedPosts(AutoGeneratedRequest) returns (AutoGeneratedResponse);
    rpc ManualGeneratedPosts(ManualGeneratedRequest) returns (ManualGeneratedResponse);
    rpc AnalyticPosts(AnalyticPostsRequest) returns (AnalyticPostsResponse);
}

// Auto post

message AutoGeneratedRequest {

    repeated AutoPost posts = 1 ;

    string prompt_base = 2;
    string tone_of_voice = 3;
    string lang = 4;
    repeated string prompts = 5;


    uint32 posted_at = 6;
    uint32 words_count = 7;

    bool use_emoji = 8;
    bool use_hashtag = 9;
}

message AutoPost{
    uint32 id = 1;
    string content = 2;
    string posted_at=3;
}

message AutoGeneratedResponse {
    uint32 status_code = 1;
    string detail = 2;
    repeated AutoData data = 3;
}

message AutoData {
    uint32 parent_id = 1;
    string content = 2;
    string scheduled_at =3;
}

// Manual post

message ManualGeneratedRequest{
    string prompt = 1 ;
    string lang = 2;
}
message ManualGeneratedResponse{
    uint32 status_code = 1;
    string detail = 2;
    string data = 3;
}

// Analytic

message AnalyticPostsRequest{
    repeated AnalyticPost posts = 1;
    string lang = 2;
}

message AnalyticPost{
    uint32 id = 1;
    uint32 channel = 2;
    string content = 3;
    string posted_at = 4;
    repeated string text_urls = 5;
    uint32 views = 6;
    uint32 forwards = 7;
    uint32 replies_count = 8;
    uint32 total_reactions = 9;
    uint32 stars = 10;
    repeated ReactionBreakdown reaction = 11;
}

message ReactionBreakdown{
    uint32 count = 1;
    string emoji = 2;
}

message AnalyticPostsResponse{
    uint32 status_code = 1;
    string detail = 2;
    repeated AnalyticData data = 3;
}

message AnalyticData {
    uint32 post_id = 1;
    uint32 post_channel = 2;
    float er_score = 3;
    float sentiment_score = 4;
    float final_score = 5;
    string posted_at = 6;
}